### PR TITLE
Allow callback of initCommand to accept multiple parameters.

### PR DESCRIPTION
**Pull request by changsi-an**
Addresses #195

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -280,7 +283,7 @@ function initAsyncCommand<T>(extensionContext: vscode.ExtensionContext, commandI
             if (args.length === 0) {
                 await callback();
             } else {
-                await callback(<T>args[0]);
+                await callback.apply(null, args);
```
Comment (EricJizbaMSFT) [inline src/extension.ts]: I'm not sure setting `this` to `null` will work in all cases? What if the callback has a different value they want for `this`?

File: src/extension.ts [hunk 2]
```diff
@@ -257,18 +257,21 @@ export function activate(context: vscode.ExtensionContext): void {
...
 }
 
 // tslint:disable-next-line:no-empty
 export function deactivate(): void {
 }
 
-function initCommand<T>(extensionContext: vscode.ExtensionContext, commandId: string, callback: (context?: T) => void): void {
-    initAsyncCommand(extensionContext, commandId, async (context?: T) => callback(context));
+// tslint:disable-next-line:no-any
+function initCommand<T>(extensionContext: vscode.ExtensionContext, commandId: string, callback: (context?: T, ...rest: any[]) => void): void {
```
Comment (EricJizbaMSFT) [inline src/extension.ts]: It doesn't seem like the type parameter `T` is providing much value anymore and is just adding complexity

What if we just simplify this to be:
```
function initCommand(context: vscode.ExtensionContext, commandId: string, callback: (...args: any[]) => any)
```

Comment (EricJizbaMSFT) [reply src/extension.ts]: And then you can just call `await callback(...args);` down below

The cosmos repo has the simplified signature of `initCommand` and `initAsyncCommand`
https://github.com/Microsoft/vscode-cosmosdb/blob/master/src/extension.ts#L145

Comment (changsi-an) [reply src/extension.ts]: I thought about that too. Not sure if there would be any special purpose of type T, I just wanted to minimize the impact to the codebase as much as possible.

Will change it in another PR.

File: src/extension.ts (hunk 1 above)
Comment (changsi-an) [reply src/extension.ts]: I guess that the caller has to remember to use arrow functions all the time. Otherwise, is that anyway to maintain that?

Comment (mike-kaufman) [reply src/extension.ts]: `callback.apply(null, args)` is equivalent to `callback(...args)`.  If caller wants a specific this context when the callback is invoked, they need to do the binding in the callback.  e.g.,

```
callback = function() {
    myFunc.apply(myThis, args);
}
```

Comment (changsi-an) [reply src/extension.ts]: Yeah, was wondering how TS transcompile would avoid this problem.

I guess after this change, it requires all callers to remember always use arrow functions as callback to preserve `this`. Does it make sense?

Comment (mike-kaufman) [reply src/extension.ts]: Yeah, either arrow functions, or explicitly bound functions via [Function's bind() method](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/bind), or equivalent as above.

**Pull request merged**